melosys-eessi-app/src/main/java/no/nav/melosys/eessi/config/KafkaAivenConfig.java

```diff
@@ -0,0 +1,139 @@
+        KafkaProperties kafkaProperties, @Value("${melosys.kafka.aiven.consumer.groupid}") String groupId) {
+        return kafkaListenerContainerFactory(SedHendelse.class, kafkaProperties, groupId);
+    }
+
+    public <T> ConcurrentKafkaListenerContainerFactory<String, T> kafkaListenerContainerFactory(
+        Class<T> containerType, KafkaProperties kafkaProperties, String groupId) {
+
+        Map<String, Object> props = kafkaProperties.buildConsumerProperties();
+        props.putAll(consumerConfig(groupId));
+        DefaultKafkaConsumerFactory<String, T> defaultKafkaConsumerFactory = new DefaultKafkaConsumerFactory<>(
+            props, new StringDeserializer(), valueDeserializer(containerType));
+        ConcurrentKafkaListenerContainerFactory<String, T> factory = new ConcurrentKafkaListenerContainerFactory<>();
+        factory.setConsumerFactory(defaultKafkaConsumerFactory);
+
+        return factory;
+    }
+
+    private Map<String, Object> producerProps() {
+        Map<String, Object> props = new HashMap<>();
+
+        props.put(CommonClientConfigs.CLIENT_ID_CONFIG, "eessi-producer");
+        props.put(ProducerConfig.ACKS_CONFIG, "all");
+        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, brokersUrl);
+
+        if (isNotLocal()) {
+            props.putAll(nonLocalSecurityConfig());
+        }
+        return props;
+    }
+
+    private Map<String, Object> consumerConfig(String groupId) {
+        Map<String, Object> props = new HashMap<>();
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, groupId);
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokersUrl);
+        props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        props.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, 15000);
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, JsonDeserializer.class);
+        props.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1);
+
+        if (isNotLocal()) {
+            props.putAll(nonLocalSecurityConfig());
+        }
+
+        return props;
+    }
+
+    private Map<String, Object> nonLocalSecurityConfig() {
+        Map<String, Object> props = new HashMap<>();
+        props.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SSL");
+
+        props.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, truststorePath);
+        props.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, credstorePassword);
+        props.put(SslConfigs.SSL_TRUSTSTORE_TYPE_CONFIG, "JKS");
+
+        props.put(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, keystorePath);
+        props.put(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, credstorePassword);
+        props.put(SslConfigs.SSL_KEY_PASSWORD_CONFIG, credstorePassword);
+        props.put(SslConfigs.SSL_KEYSTORE_TYPE_CONFIG, "PKCS12");
+
+        return props;
+    }
```

Comment:
Samme spørsmål her om konstantene vi hardkoder bør tas ut til properties eller ikke

Reply:
Har kopiert dette fra melosys-api. Såvidt jeg har skjønt så har vi ikke noen konstanter i properties med mindre de er miljøavhengige. 
Har spurt om dette tidligere før selv. Svaret jeg fikk var at det ikke lenger spiller noen rolle hvorvidt det er hardkodet eller i properties. Tidligere kunne man endre ting i properties i produksjon, uten å lage nye pakker. I dag er det enklere å endre på kode og få ut ny versjon i stedet for å endre på properties-filene i pods.